src/pveclib/vec_int128_ppc.h

```diff
@@ -903,6 +1059,107 @@ example_longdiv_10e31 (vui128_t *q, vui128_t *d, long int _N)
  * <I>Latency</I> and <I>Throughput</I> are derived see:
  * \ref perf_data
  */
+
+/** \brief Generate a vector unsigned __int128 constant from words.
+ *
+ *  Combine 4 x 32-bit int constants into a single __int128 constant.
+ *  The 4 parameters are integer constant values in high to low order.
+ *  This order is consistent for big and little endian and the result
+ *  loaded into vector registers is correct for quadword  integer
+ *  operations.
+ *
+ *  For example
+ * \code
+ vui128_t ten32 = CONST_VUINT128_QxW (0x000004ee, 0x2d6d415b, 0x85acef81, 0x00000000);
+ * \endcode
+ *
+ */
+#define CONST_VUINT128_QxW(__q0, __q1, __q2, __q3) ( (vui128_t) \
+    (((((((unsigned __int128) __q0) << 32) \
+    +    ((unsigned __int128) __q1)) << 32) \
```

Comment:
Ya this is tricky as the (unsigned __int128) cast must be applied to each of the 4 const, but before the shift of the preceding term.

I looked at adding white space to match to levels:
#define CONST_VUINT128_QxW(__q0, __q1, __q2, __q3) ( (vui128_t) \
    (( (( (((unsigned __int128) __q0) << 32) \
    +      ((unsigned __int128) __q1) ) << 32) \
    +      ((unsigned __int128) __q2) ) << 32) \
    +      ((unsigned __int128) __q3))

Bu I am not convinced this helps ....

Reply:
Added the comment:

 *  The effect is to compute an unsigned __int128 constant from 4 x
 *  32-bit unsigned int constants.
 * \code
 * int128 = (((((__q0 << 32) + __q1) << 32) + __q2) << 32) + q3
 * \endcode

Reply:
I agree with @ThinkOpenly's comment here (that somehow ended up separate from mine):

> showing my ignorance... do you need all the nesting here? can you shift by 96 and 64? If not, should you make note of that in a comment?

Since you're doing the cast to full width on each `q` individually... I think shifting by 96, 64, and then 32 would be clearer than shifting and summing this way.